active/CornerRadius/CornerRadiusFilterConverter.md

```diff
@@ -0,0 +1,63 @@
+# Background
+
+Xaml controls can have rounded corners, usually controlled by setting the CornerRadius property on [Control](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Control.CornerRadius) or [Border](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Border.CornerRadius). The [CornerRadius](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.CornerRadius) type itself is a struct with fields TopLeft, TopRight, BottomLeft, and BottomRight.
+
+Typically, a control's ControlTemplate binds the control's CornerRadius to an item in the template, for example this sets the CornerRadius to (5,5,5,5):
+
+```
+<Button Background="Gray" Padding="5" CornerRadius="5">
+    Click
+    <Button.Template>
+        <ControlTemplate TargetType="Button">
+            <Border Background="{TemplateBinding Background}"
+                    CornerRadius="{TemplateBinding CornerRadius}"
+                    Padding="{TemplateBinding Padding}">
+                <ContentPresenter />
+            </Border>
+        </ControlTemplate>
+    </Button.Template>
+</Button>
+```
+
+But there are cases where the ControlTemplate needs to get only some of the components out of the CornerRadius struct, for example the top left and right corners, in order to have a rounded top and a square bottom. There's no way to create a TemplateBinding to a struct field, so you can't bind to CornerRadius.TopLeft.
+
+For example when CommandBarFlyout opens up its secondary menu, primary and secondary menus are clipped together we need to clear rounded corners on bottom of primary menu and top of secondary menu to close the gaps.
+
+![CommandBarFlyout CornerRadius example](images/commandbarflyout-cornerradius-example.png)
+
+The solution in this spec is a stock IValueConverter that can be used with the TemplateBinding to extract fields out of the struct.
+
+# Description
+
+CornerRadiusFilterConverter helps us filter CornerRadius values on specific directions and set values to 0 for the rest.
+
+# Examples
+A ControlTemplate for a Button that has square bottom corners, and top corners that can be set by the Button.CornerRadius property.
+```XAML
+<ControlTemplate TargetType="Button">
+    <Border Background="{TemplateBinding Background}"
+            CornerRadius="{TemplateBinding CornerRadius, Converter={StaticResource CornerRadiusFilterConverter}, ConverterParameter=Top}"

```

Comment:
The bad thing about using ConverterParameter is that it's untyped, so you don't know what's valid (no Intellisense support or compile-time checking). Better would be to make the parameter a property on the converter class. So:

```
<Grid.Resources>
    <CornerRadiusFilterConverter Filter="Top" x:Name="TopCornersConverter" />
</Grid.Resources>
...
CornerRadius="{TemplateBinding CornerRadius, Converter={StaticResource TopCornersConverter}}"
...
enum CornerRadiusFilterKind
{
    Top,
    Right,
    Left,
    Bottom
}
```

Reply:
@MikeHillberg I like this suggestion, but what would you recommend for the key names that we put in the resource dictionary? TopCornerRadiusFilterConverter, LeftCornerRadiusFilterConverter, etc? "TopCornersConverter" seems a little too vague for a global key.

Reply:
This is going to be a global resource (generic.xaml)?

Reply:
Yes, since it would need to be used by default styles of controls.